examples/fips/README.md

```diff
@@ -0,0 +1,83 @@
+
+
+### Install from source by building librdkafka
+
+When building librdkafka and python client from source, librdkafka uses OpenSSL present in the system by dynamically linking it. If the installed OpenSSL is already working in FIPS mode, then you can directly jump to section [Client configuration to enable FIPS provider](#client-configuration-to-enable-fips-provider), otherwise use steps mentioned in section [Enabling FIPS provider](#enabling-fips-provider) to make system OpenSSL FIPS compliant. Once that is done, librdkafka and python client will use FIPS approved algorithms for the communication between client and Kafka Cluster using producer, consumer or admin client.
+
+### Enabling FIPS provider
+
+We assume that you have FIPS provider (module) available in your system. This module can be dynamically plugged into OpenSSL by putting this module in default module folder of OpenSSL or pointing to this module with environment vairable `OPENSSL_MODULES` (folder containing the FIPS module in the form of dynamic library). We also need to use OpenSSL config which has FIPS configuration present in it. Default configuration file can be modified to include FIPS related config or a new configuration file can also be pointed using `OPENSSL_CONF` environment variable.
+
+#### Steps to generate FIPS provider
+
+Steps to generate FIPS provider can be found on the [README-FIPS doc](https://github.com/openssl/openssl/blob/openssl-3.0.8/README-FIPS.md)
+
+In short, you need to perform the following steps:
+
+1) Clone OpenSSL from [OpenSSL Github Repo](https://github.com/openssl/openssl)
+2) Checkout the correct version. (v3.0.8 is the current FIPS compliant version for OpenSSL 3.0 at the time of writing this doc.)
+3) Run `./configure --enable-fips`
+4) Run `make install_fips`
+
+After last step, two files are generated.
+* FIPS module (`fips.dylib` in Mac, `fips.so` in Linux and `fips.dll` in Windows)
+* FIPS config (`fipsmodule.cnf`) file will be generated. Which needs to be used with OpenSSL.
+
+#### Referencing FIPS provider in OpenSSL
+
+As mentioned earlier, the above generated FIPS module can be dynamically plugged into OpenSSL by putting this module in default module folder (Refer [this](https://github.com/confluentinc/librdkafka/blob/master/INTRODUCTION.md#ssl) for typical default locations on various OS) of OpenSSL. It should look something like `...lib/ossl-modules/`.
+
+This module can also be pointed with environment variable `OPENSSL_MODULES` (folder containing the FIPS module in the form of dynamic library)
+
+#### Linking FIPS provider with OpenSSL
+
+`fipsmodule.cnf` file includes `fips_sect` which is required to enable FIPS in OpenSSL. `fipsmodule.cnf` needs to be included in `openssl.cnf` file to make it work.
```

Comment:
```suggestion
To enable FIPS in OpenSSL, you must include `fipsmodule.cnf` in the file, `openssl.cnf`. The `fipsmodule.cnf` file includes `fips_sect` which OpenSSL requires to enable FIPS. See the example below. 
```
I struggled to understand this part. I am happy to look at it again. The main thing is to make clear what the customer has to do to get it to work.

Reply:
Used suggestion. It looks fine.